docs/csharp/language-reference/builtin-types/numeric-conversions.md

```diff
@@ -0,0 +1,105 @@
+
+  - If the source type is larger than the destination type, then the source value is truncated by discarding its "extra" most significant bits. The result is then treated as a value of the destination type.
+
+  - If the source type is smaller than the destination type, then the source value is either sign-extended or zero-extended so that it's of the same size as the destination type. Sign-extension is used if the source type is signed; zero-extension is used if the source type is unsigned. The result is then treated as a value of the destination type.
+
+  - If the source type is the same size as the destination type, then the source value is treated as a value of the destination type.
+
+- When you convert a `decimal` value to an integral type, this value is rounded towards zero to the nearest integral value. If the resulting integral value is outside the range of the destination type, an <xref:System.OverflowException> is thrown.
+
+- When you convert a `double` or `float` value to an integral type, this value is rounded towards zero to the nearest integral value. If the resulting integral value is outside the range of the destination type, the result depends on the overflow [checking context](../keywords/checked-and-unchecked.md). In a checked context, an <xref:System.OverflowException> is thrown, while in an unchecked context, the result is an unspecified value of the destination type.
+
+- When you convert `double` to `float`, the `double` value is rounded to the nearest `float` value. If the `double` value is too small or too large to fit into the `float` type, the result is zero or infinity.
+
+- When you convert `float` or `double` to `decimal`, the source value is converted to `decimal` representation and rounded to the nearest number after the 28th decimal place if required. Depending on the value of the source value, one of the following results may occur:
+
+  - If the source value is too small to be represented as a `decimal`, the result becomes zero.
+
+  - If the source value is NaN (not a number), infinity, or too large to be represented as a `decimal`, an <xref:System.OverflowException> is thrown.
+
+- When you convert `decimal` to `float` or `double`, the source value is rounded to the nearest `float` or `double` value, respectively.
+
+## C# language specification
+
+For more information, see the following sections of the [C# language specification](~/_csharplang/spec/introduction.md):
+
+- [Implicit numeric conversions](~/_csharplang/spec/conversions.md#implicit-numeric-conversions)
+- [Explicit numeric conversions](~/_csharplang/spec/conversions.md#explicit-numeric-conversions)
+
+## See also
+
+- [C# reference](../index.md)
+- [Casting and type conversions](../../programming-guide/types/casting-and-type-conversions.md)
```

Comment:
```suggestion
- [Casting and type conversions](../../programming-guide/types/casting-and-type-conversions.md)
- [Compiler error CS0221](../../misc/cs0221.md)
```

Reply:
@Youssef1313 what makes you think such a link would be useful here? In general, I don't consider the compiler error pages as a reference material for occasional exploration, but only as a reference when one faces the error in one's code.

Reply:
I thought it would be useful as this error message was mentioned in this line:

```csharp
byte b = 300;   // CS0031: Constant value '300' cannot be converted to a 'byte'
```

Reply:
I referenced another error instead of CS0031 by mistake

Reply:
@Youssef1313 thanks! I've added the link to the compiler error page to the new paragraph that follows the snippet. I'm resolving this conversation, then. Please check the latest commit.

> I referenced another error instead of CS0031 by mistake

By the way, in general, when you edit your comment, you should be able to edit the suggestion that comment contains.